src/components/forms/text-input/text-input.tsx

```diff
@@ -66,6 +71,7 @@ export const TextInput = (
       <Input
         type={'text'}
         aria-required={true}
+        aria-invalid={Boolean(props.errors?.[props.name])}
```

Comment:
I would reuse the existing error logic here and in the other input components for two reasons:
1. The visual error handling and the error handling for screen readers are consistent.
2. You don't need any changes in the parent and no new logic in the input component 
```suggestion
        aria-invalid={Boolean(errorMessage)}
```